### PR TITLE
Fix Memory Issues and Optimizations in Code Usage 

### PR DESCRIPTION
**Pull request by samirsahu1947**
In this scenario,  if realloc fails, then the original memory allocation exists, but is leaked as it is no longer referred. So you need to keep a handle on the original allocation until you have verified that the new allocation is valid.

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check Result:
os/logm/logm_process.c:47: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/logm/logm_process.c:47: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/logm/logm_process.c:50: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/logm/logm_process.c:50: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by jeongarmy** [comment]
additionally, else syntax is not nessasary because it will exit when error

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

File: os/logm/logm_process.c [hunk 1]
```diff
@@ -42,10 +42,12 @@ static int logm_change_bufsize(int buflen)
 	}
 
 	/* Realloc new buffer with new length */
-	g_logm_rsvbuf = (char *)realloc(g_logm_rsvbuf, buflen);
-	if (!g_logm_rsvbuf) {
+	char *new_g_logm_rsvbuf = (char *)realloc(g_logm_rsvbuf, buflen);
+	if (new_g_logm_rsvbuf == NULL) {
+		dbg("Realloc Fail\n");
```
Comment (jeongarmy) [inline os/logm/logm_process.c]: It is duplicated because logm task show fail message when this failed.
please see line93 in logm_process.c

Comment (samirsahu1947) [reply os/logm/logm_process.c]: For the log msg I put under dbg to specify only realloc failure
Line 93 is common for any error ....

Comment (jeongarmy) [reply os/logm/logm_process.c]: But this fail is a only failed case in this funtion.

**Comment by seinfra** [comment]
Code Rule Check OK.

File: apps/examples/fs_helper/ramfs_test.c [hunk 2]
```diff
@@ -111,6 +111,7 @@ static int make_file(char *filename)
 
 	if (fwrite(in_string, BUFFER_SIZE, 1, fp) < 0) {
 		printf("Cannot write file %s\n", filename);
+		fclose(fp);
```
Comment (Taejun-Kwon) [inline apps/examples/fs_helper/ramfs_test.c]: This is ok for me.

**Comment by seinfra** [comment]
Code Rule Check OK.

File: os/logm/logm_process.c (hunk 1 above)
Comment (sunghan-chang) [inline os/logm/logm_process.c]: How about change dbg to wdbg? I don't think this is enough important to print twice about failure. CONFIG_DEBUG can be used easily.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@samirsahu1947 Could you fill the information for module name on commit title?

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**